### DISCUSSION TITLE
Expiration of Access Token after logout

### DISCUSSION THREAD

**Discussion by santa-imascono** [directus/directus#17252] [Feature Requests]
We have noticed that **Refresh Tokens do expire** right after a user logout. However the same **does not apply to Access Tokens**. Currently, access tokens are valid until they expire regardless of the fact of the user may log out.

In terms of security, invalidating access tokens right after the user logs out would reduce the window of opportunity for an attack.

**Comment by rijkvanzanten** [comment]
That's as expected! The reason to do so is to keep the "regular" api requests stateless, which improves performance considerably. If regular access tokens would be invalidated serverside on logout as well, the API would have to check against the invalidation list for every request. You can control the TTL of the access tokens to reduce the window

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by suyashnigam50** [comment]
Hey ! In this case how can i make my api secure? i am using access token in my web application to use directus api, but even after executing the logout api i can still access the api's from the same access token. 
while in directus ui we have to login again .
